includes/results.inc

```
@@ -902,10 +902,43 @@ class IslandoraSolrFacets {
     $islandora_solr_query = self::$islandoraSolrQuery;
     $soft_limit = self::$soft_limit;
     $buckets = array();
+    $replace_bucket = (isset($this->settings['solr_field_settings']['pid_object_label']) && $this->settings['solr_field_settings']['pid_object_label'] ? TRUE : FALSE);
+    if ($replace_bucket) {
+      $query_processor = new IslandoraSolrQueryProcessor();
+    }
     foreach ($results as $key => $values) {
       $bucket = $values['bucket'];
       $filter = $values['filter'];
       $count = $values['count'];
+
+      // Replace link bucket with object label based on facet field settings.
+      if ($replace_bucket) {
+        $label = NULL;
+        $pid = str_replace('info:fedora/', '', $bucket);
+        if (islandora_is_valid_pid($pid)) {
+          try {
+            $query_processor->buildAndExecuteQuery("PID:\"$pid\"");
+            if (!empty($query_processor->islandoraSolrResult) && !empty($query_processor->islandoraSolrResult['response']['objects'])) {
+              $label = (!empty($query_processor->islandoraSolrResult['response']['objects'][0]['object_label']) ?
+              $query_processor->islandoraSolrResult['response']['objects'][0]['object_label'] : NULL);
+            }
+            // Fall back to islandora object if PID is not in solr.
+            // eg: content models.
+            else {
+              if ($object = islandora_object_load($pid)) {
+                if (islandora_object_access(ISLANDORA_VIEW_OBJECTS, $object)) {
+                  $label = $object->label;
+                }
+              }
+            }
+          }
+          catch (Exception $ex) {
```

Why a blanket try catch? Squashing errors needs a very good very well documented reason.